diff --git "a/C:\\Users\\kristina\\AppData\\Local\\Temp\\TortoiseGit\\lea94FF.tmp\\leanbackPlayer-0f314ec-left.js" "b/C:\\workspace\\hannover\\extension\\xrowvideo\\design\\standard\\javascript\\leanbackPlayer.js"
index d89f20c..d6a8825 100644
--- "a/C:\\Users\\kristina\\AppData\\Local\\Temp\\TortoiseGit\\lea94FF.tmp\\leanbackPlayer-0f314ec-left.js"
+++ "b/C:\\workspace\\hannover\\extension\\xrowvideo\\design\\standard\\javascript\\leanbackPlayer.js"
@@ -1231,7 +1231,9 @@ LBP.prototype.initializeDocumentEvents = function() {
 	try {
 		(function(p) {
 			LBP.addEvent(window, "message", function(e) {
-				var data = JSON.parse(e.data)||false;
+				var data = false;
+				try {data = JSON.parse(e.data);} catch(_ex) {}
+				
 				if(!!data && typeof(data.action) !== "undefined" && data.action === "setScreen" && typeof(data.vid) !== "undefined" && data.vid === p.options.vid && typeof(data.value) !== "undefined") {
 					p.setScreen(data.value, true, data.cors, data.bsXY);
 				} else if(!!data && typeof(data.action) !== "undefined" && data.action === "setCorsStatus" && typeof(data.vid) !== "undefined" && data.vid === p.options.vid && typeof(data.cors) !== "undefined" && typeof(data.iframe) !== "undefined") {
@@ -2882,7 +2884,7 @@ LBP.setCssStyle = function(e, s, v) {
 /* fct: get player object by id */
 LBP.getPlayer = function(id) {
 	if(id === null) {return;}
-    for(var i=0, j=_LBP_Player.length; i<j; i++) {if(id.indexOf(_LBP_Player[i].options.vid) !== -1) {return _LBP_Player[i];}}
+    for(var i=0, j=_LBP_Player.length; i<j; i++) {if(id === _LBP_Player[i].options.vid) {return _LBP_Player[i];}}
 	return;
 };
 /* fct: blur/focus player */
@@ -3010,7 +3012,8 @@ LBP.resizeToBrowser = function(h,w, bsXY) {
 };
 /* fct: receive (CORS) IFrame postMessage events */
 LBP.receiveIframeEvent = function(e) {
-	var data = JSON.parse(e.data)||false;
+	var data = false;
+	try {data = JSON.parse(e.data);} catch(_ex) {}
 	if(!!data && data.action === "setScreen") {
 		LBP.setIframeParentScreen(e);
 	} else if(!!data && data.action === "getCorsStatus") {
